cypress/e2e/09-multiplex_testing.cy.js

```diff
@@ -1,55 +1,65 @@
+const currentSpecRunVersionName = `${testNumber()}-cypress-${specRunName}`;
 
 loginHooks();
 describe("Testing with multiplex devices", () => {
-  let patient, facility, multiplexDeviceName;
+  let facilityId, patientId;
+  const patient = generatePatient();
+  const multiplexDevice = generateMultiplexDevice();
 
   before(() => {
-    cy.makePOSTRequest({
-      operationName: "WhoAmI",
-      variables: {},
-      query:
-        "query WhoAmI {\n  whoami {\n organization {\n    facilities {\n      id\n      name\n      __typename\n    }\n    __typename\n  }\n} \n}",
-    }).then((res) => {
-      facility = res.body.data.whoami.organization.facilities[0];
-      cy.makePOSTRequest({
-        operationName: "GetPatientsByFacility",
-        variables: {
-          facilityId: facility.id,
-          pageNumber: 0,
-          pageSize: 1,
-          archivedStatus: "UNARCHIVED",
-        },
-        query:
-          "query GetPatientsByFacility($facilityId: ID!, $pageNumber: Int!, $pageSize: Int!, $archivedStatus: ArchivedStatus, $namePrefixMatch: String) {\n  patients(\n    facilityId: $facilityId\n    pageNumber: $pageNumber\n    pageSize: $pageSize\n    archivedStatus: $archivedStatus\n    namePrefixMatch: $namePrefixMatch\n  ) {\n    internalId\n    firstName\n    lastName\n    middleName\n    birthDate\n    isDeleted\n    role\n    lastTest {\n      dateAdded\n      __typename\n    }\n    __typename\n  }\n}",
-      }).then((res) => {
-        patient = res.body.data.patients[0];
-      });
-    });
+    cy.task("getSpecRunVersionName", specRunName).then(
+      (prevSpecRunVersionName) => {
+        if (prevSpecRunVersionName) {
+          cleanUpPreviousRunSetupData(prevSpecRunVersionName);
+          cleanUpRunOktaOrgs(prevSpecRunVersionName);
+        }
+        let data = {
+          specRunName: specRunName,
+          versionName: currentSpecRunVersionName,
+        };
+        cy.task("setSpecRunVersionName", data);
 
-    cy.task("getMultiplexDeviceName").then((name) => {
-      multiplexDeviceName = name;
-    });
+        setupRunData(currentSpecRunVersionName).then((result) => {
+          facilityId = result.body.data.addFacility.id;
+        });
+        setupPatient(currentSpecRunVersionName, patient).then((result) => {
+          patientId = result.internalId;
+        });
+        setupMultiplexDevice(currentSpecRunVersionName, multiplexDevice);
+      },
+    );
   });
 
   beforeEach(() => {
     cy.intercept("POST", graphqlURL, (req) => {
       aliasGraphqlOperations(req);
     });
+  });
 
-    // remove a test for the patient if it exists
-    cy.makePOSTRequest({
-      operationName: "RemovePatientFromQueue",
-      variables: { patientId: patient.internalId },
-      query:
-        "mutation RemovePatientFromQueue($patientId: ID!) {\n  removePatientFromQueue(patientId: $patientId)\n}",
-    });
+  after("clean up spec data", () => {
+    cleanUpPreviousRunSetupData(currentSpecRunVersionName);
```

Comment:
Do we need to add the device to the cleanup actions here? Currently the device artifacts are left behind on each run

Reply:
Are there any harmful side effects to not cleaning up these devices in the tests themselves? Each run will have a unique device and facility so I don't think this would cause conflicts with other specs. I think the upcoming changes to our database seeding in #7236 will make it a lot easier to wipe and reset your local db if things start to feel bloated with test data

Reply:
none that i can think of! i think that a better db wipe should work ok for local dev